dlib/dnn/layers.h

```diff
@@ -1522,6 +1522,36 @@ namespace dlib
 
             unsigned long new_window_size;
         };
+
+        class visitor_bn_input_no_bias
+        {
+        public:
+
+            template <typename T>
+            void set_input_no_bias(T&) const
+            {
+                // ignore other layer types
+            }
+
+            template <layer_mode mode, typename U, typename E>
+            void set_input_no_bias(add_layer<bn_<mode>, U, E>& l)
+            {
+                l.subnet().layer_details().set_bias_learning_rate_multiplier(0);
+                l.subnet().layer_details().set_bias_weight_decay_multiplier(0);
```

Comment:
This should be this instead right?
```
        set_bias_learning_rate_multiplier(l.subnet().layer_details(), 0);
        set_bias_weight_decay_multiplier(l.subnet().layer_details(), 0);
```
just in case someone sticks a batch norm layer on top of something that doesn't have a `.set_bias_learning_rate_multiplier`

Reply:
Oh, good catch! thank you :)